app/code/Magento/Customer/Model/ResourceModel/Address/Attribute/Source/CountryWithWebsites.php

```diff
@@ -117,17 +117,10 @@ public function getAllOptions($withEmpty = true, $defaultValues = false)
 
                 $allowedCountries = array_unique(array_merge([], ...$allowedCountries));
             } else {
-                // Address can be added only for the allowed country list.
-                $websiteId = null;
-                $customerId = $this->request->getParam('parent_id') ?? null;
-                if ($customerId) {
```

Comment:
The removed code checked for parent_id parameter and loaded customer-specific website. This may have been intentional for admin address editing. Need to verify if this change affects admin panel functionality.